python/lsst/obs/base/_fitsRawFormatterBase.py

```diff
@@ -350,41 +331,20 @@ def readComponent(self, component, parameters=None):
             detector = self.getDetector(self.observationInfo.detector_num)
             visitInfo = self.makeVisitInfo()
             return self.makeWcs(visitInfo, detector)
+        elif component == "metadata":
+            self.stripMetadata()
+            return self.metadata
         return None
 
-    def readFull(self, parameters=None):
-        """Read the full Exposure object.
-
-        Parameters
-        ----------
-        parameters : `dict`, optional
-            If specified, a dictionary of slicing parameters that overrides
-            those in the `fileDescriptor` attribute.
-
-        Returns
-        -------
-        exposure : `~lsst.afw.image.Exposure`
-            Complete in-memory exposure.
-        """
-        from lsst.afw.image import makeExposure, makeMaskedImage
-        full = makeExposure(makeMaskedImage(self.readImage()))
-        mask = self.readMask()
-        if mask is not None:
-            full.setMask(mask)
-        variance = self.readVariance()
-        if variance is not None:
-            full.setVariance(variance)
+    def readFull(self):
+        # Docstring inherited.
+        self.checked_parameters  # just for checking; no supported parameters.
+        full = lsst.afw.image.makeExposure(lsst.afw.image.makeMaskedImage(self.readImage()))
         full.setDetector(self.getDetector(self.observationInfo.detector_num))
-        info = full.getInfo()
-        info.setFilterLabel(self.makeFilterLabel())
-        info.setVisitInfo(self.makeVisitInfo())
-        info.setWcs(self.makeWcs(info.getVisitInfo(), info.getDetector()))
-        # We don't need to call stripMetadata() here because it has already
-        # been stripped during creation of the ObservationInfo, WCS, etc.
-        full.setMetadata(self.metadata)
+        self.attachComponentsFromMetadata(full)
         return full
 
-    def readRawHeaderWcs(self, parameters=None):
+    def readRawHeaderWcs(self):
         """Read the SkyWcs stored in the un-modified raw FITS WCS header keys.
         """
         return lsst.afw.geom.makeSkyWcs(lsst.afw.fits.readMetadata(self.fileDescriptor.location.path))
```

Comment:
It's a bit annoying that we have to read the header twice just because we might have stripped the header by calling stripMetadata and dropping the resulting WCS object on the floor.  Also note that if we are going to read the header again we have to apply fix_header here to ensure that we have applied corrections that might affect the WCS. Can we guarantee that this routine is only ever called if stripMetadata has not been called (because it's only called if a component is being read)? Should we cache the stripped and unstripped headers?

Reply:
This is something I moved around but tried not to change, but now that you mention it, I don't see this method called anywhere, even to satisfy a component.  I am inclined to delete it and see if Jenkins complains.

Reply:
I think this method exists because in gen2 we have a butler component to return the header wcs. We haven't enabled that feature in the gen3 as a derived component (because initially we hadn't got derived components). I think the component is called `header_wcs` in gen2. RFC-616 I think.

Reply:
This may be beyond relevance, but we had a previous discussion about this on DM-24024, specifically on this and the following three comments: https://jira.lsstcorp.org/browse/DM-24024?focusedCommentId=261332&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-261332

Reply:
@laurenam, I was actually going to ask if you thought you'd notice if Gen3 never got around to providing `raw.header_wcs`.  I'm currently thinking that if nobody has noticed that we don't have it yet, maybe we never should, as the kind of problems it was helpful to debug are just not problems anymore, now that Gen3 is more consistent about the raw WCS being the right one.

Reply:
I’m inclined to agree with your current line of thinking...